### PR TITLE
feat: show app age in application list view (#11209)

### LINKED ISSUE #11209: feat(ui): show application's age in the application list view
# Summary

Show application age (time since created) in the application list view.

# Motivation

While dealing with a large number of applications (and some of them short lived), it would be useful to see when the application was created (or what age does it have in the cluster).

This is only possible atm by clicking on a panel and checking the root node of the application details view.

# Proposal

This could be implemented as a label on the card, like in this mockup:

Screenshot 2022-11-06 at 16 01 14

### PR DESCRIPTION
**Pull request by alexef**
closes: #11209

Screenshot 2022-11-30 at 07 36 15
Screenshot 2022-11-30 at 09 57 17

Signed-off-by: Alex Eftimie <alex.eftimie@getyourguide.com>

Note on DCO:

If the DCO action in the integration test fails, one or more of your commits are not signed off. Please click on the *Details* link next to the DCO action for instructions on how to resolve this.

Checklist:

* [ ] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo-cd/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this does not need to be in the release notes.
* [ ] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] I've included "Closes [ISSUE #]" or "Fixes [ISSUE #]" in the description to automatically close the associated issue.
* [ ] I've updated both the CLI and UI to expose my feature, or I plan to submit a second PR with them.
* [ ] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [ ] Optional. My organization is added to USERS.md.
* [ ] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/tree/master/community#contributing-to-argo)
* [ ] I have written unit and/or e2e tests for my change. PRs without these are unlikely to be merged.
* [ ] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/latest/developer-guide/ci/)).

### CODE REVIEW

File: ui/src/app/applications/components/applications-list/applications-tiles.tsx [hunk 1]
```diff
@@ -256,6 +257,16 @@ export const ApplicationTiles = ({applications, syncApplication, refreshApplicat
                                                         </div>
                                                         <div className='columns small-9'>{app.spec.destination.namespace}</div>
                                                     </div>
+                                                    <div className='row'>
+                                                        <div className='columns small-3' title='Age:'>
+                                                            Age:
+                                                        </div>
```
Comment (ashutosh16) [inline ui/src/app/applications/components/applications-list/applications-tiles.tsx:263]: In the application summary, we use the label `CREATED AT` to show when the resource was created. I think we should be consistent with the label. Would it be okay to change the label `Age` to `Created AT`

File: ui/src/app/applications/components/applications-list/applications-table.tsx [hunk 2]
```diff
@@ -118,22 +119,32 @@ export const ApplicationsTable = (props: {
...
-                                                            <i className='fa fa-ellipsis-v' />
-                                                        </button>
-                                                    )}
-                                                    items={[
-                                                        {title: 'Sync', action: () => props.syncApplication(app.metadata.name, app.metadata.namespace)},
-                                                        {title: 'Refresh', action: () => props.refreshApplication(app.metadata.name, app.metadata.namespace)},
-                                                        {title: 'Delete', action: () => props.deleteApplication(app.metadata.name, app.metadata.namespace)}
-                                                    ]}
-                                                />
+                                                <div className='row'>
```
Comment (ashutosh16) [inline ui/src/app/applications/components/applications-list/applications-table.tsx:124]: I feel the table has too many columns to display and it's slightly cramped, I would prefer to have created time in the tooltip
Screen Shot 2022-11-30 at 12 23 54 PM

File: ui/src/app/applications/components/applications-list/applications-tiles.tsx (hunk 1 above)
Comment (alexef) [reply ui/src/app/applications/components/applications-list/applications-tiles.tsx:263]: But `Created at` is a timestamp, while Age is a time delta (as in the screenshot, x hours days etc).

In the tree view, the root node shows the age (and nodes in general have an age label), which makes it easier to observe the resources - as you don't have to do the math now-created_at.

File: ui/src/app/applications/components/applications-list/applications-table.tsx (hunk 2 above)
Comment (alexef) [reply ui/src/app/applications/components/applications-list/applications-table.tsx:124]: I agree with you, but where exactly in the tooltip? or you mean as a tooltip of one table row?

Comment (ashutosh16) [reply ui/src/app/applications/components/applications-list/applications-table.tsx:124]: I guess the tooltip on the table row with additional metadata would be better however if you just want to display `age` then I would prefer tooltip on the application name

File: ui/src/app/applications/components/applications-list/applications-tiles.tsx (hunk 1 above)
Comment (ashutosh16) [reply ui/src/app/applications/components/applications-list/applications-tiles.tsx:263]: In the pod summary we display like this `CREATED AT
12/01/2022 07:15:48 (2 hours ago)`. Can we use this label and format?

Comment (alexef) [reply ui/src/app/applications/components/applications-list/applications-tiles.tsx:263]: yes, I'll do that

**Comment by alexef** [comment]
@ashutosh16 updated according to your comments
Screenshot 2022-12-02 at 15 07 29
Screenshot 2022-12-02 at 15 07 42

**Comment by ashutosh16** [comment]
Verified the changes in my local. LGTM

**Review by ashutosh16 — APPROVED** [review]
LGTM

**Review by rbreeze — APPROVED** [review]
LGTM!

**Pull request merged**